app/scripts/components/modelcontrolmenu.js

```diff
@@ -101,12 +100,41 @@ var exports = (function () {
         // Show the dialog:
         this.deleteDialog.show();
       },
-
+      someSelectedModelsAreFinished: function () {
+        return _.values(store.getSelectedModels()).some(function(model) {
+          return model.data.state === "Finished";
+        });
+      },
+      someSelectedModelsArePublished: function () {
+        return _.values(store.getSelectedModels()).some(function(model) {
+          return model.data.shared !== "p";
+        });
+      },
+      someSelectedModelsAreAlreadyPublished: function (domain) {
```

Comment:
The function name is almost as long as the implementation, it should be a separate [service](https://api.left-pad.io/?len=10&str=padded).